### PR TITLE
rpmbuild: make_srpmbuild, set recursive safe.directory

### LINKED ISSUE #3421: The make_srpm method fails with git safe.directory error
Full log: [builder-live.log.gz.txt](https://github.com/user-attachments/files/17113031/builder-live.log.gz.txt)

```
./contrib/fedora/make_srpm.sh --prerelease --output /mnt/safe-resultdir-c3dy38x1
remote: fatal: detected dubious ownership in repository at '/mnt/workdir-a7gx95j6/sssd/.git'        
remote: To add an exception for this directory, call:        
remote: 
remote: 	git config --global --add safe.directory /mnt/workdir-a7gx95j6/sssd/.git        
remote: git upload-archive: archiver died with error
fatal: sent error to the client: git upload-archive: archiver died with error
```

But strangely enough, after the fatal error, we still produce the SRPM?

```
Creating SRPM without extra patches.
setting SOURCE_DATE_EPOCH=1611187200
Wrote: /mnt/workdir-a7gx95j6/sssd/rpmbuild/SRPMS/sssd-2.10.0~beta2-0.240924.083430.git5f5077ac1.src.rpm
```

### PR DESCRIPTION
**Pull request by nikromen**
Relates #3421

### CODE REVIEW

File: rpmbuild/copr_rpmbuild/providers/scm.py [hunk 1]
```diff
@@ -17,7 +17,7 @@
 MAKE_SRPM_TEPMLATE = (
     "set -x && "
     'cd {0} && '
-    'echo -e "[safe]\ndirectory = {4}" > ~/.gitconfig && '
+    'echo -e "[safe]\ndirectory = {4}/*" > ~/.gitconfig && '
```
Comment (praiskup) [inline rpmbuild/copr_rpmbuild/providers/scm.py:20]: +1

File: rpmbuild/copr_rpmbuild/providers/scm.py [hunk 2]
```diff
@@ -118,7 +118,7 @@ def get_make_srpm_command(self):
 
         makefile_path = os.path.join(mock_repodir, '.copr', 'Makefile')
         make_srpm_cmd_part = MAKE_SRPM_TEPMLATE.format(mock_cwd, makefile_path,
-                mock_resultdir, mock_spec_path, mock_repodir)
+                mock_resultdir, mock_spec_path, mock_workdir)
```
Comment (praiskup) [inline rpmbuild/copr_rpmbuild/providers/scm.py:121]: This is weird though; the mock_repodir should be the directory containing `.git` subdir, right?

Comment (nikromen) [reply rpmbuild/copr_rpmbuild/providers/scm.py:121]: that's just remnant to make it work. Actually I want mock_repodir and do:
```
[safe]
  directory = {4}
  directory = {4}/*
```

**Comment by nikromen** [comment]
@praiskup OK this was reason why we moved repo root as safe directory - https://github.com/fedora-copr/copr/issues/2597#issuecomment-1487901896 so I don't think we should revert this change. We probably want the `/*` feature

**Comment by praiskup** [comment]
I still don't get why :sweat_smile: I experimented with permissions like:

    # I'm "praiskup"
    $ cd /tmp
    $ git clone ~/rh/projects/copr/copr
    $ cd /tmp/copr
    # sudo su
    # chown postgres:postgres cli
    # git status
    On branch main
    Your branch is up to date with 'origin/main'.

    nothing to commit, working tree clean
    # chown postgres:postgres /tmp/copr
    # git status
    fatal: detected dubious ownership in repository at '/tmp/copr'
    To add an exception for this directory, call:

            git config --global --add safe.directory /tmp/cop

IOW, there's no issue with the git tree sub-directory.  Only if the top-level
directory is mis-owned.  But I agree we don't want to revert the change;
the problem there clearly fixes the problem if subdirectory is used as
`cwd`.

**Comment by praiskup** [comment]
We talked about this PR on f2f meeting, and while the `<dir>` to `<dir>/*` change is not necessarily needed now (a different fix was needed), some users in the future might consider it useful (they might be doing manual git clone in sub-dirs, and there's no need to bother them with tweaking the safe.directory configuration).

**Comment by praiskup** [comment]
Jirka promised to update the commit message.

**Comment by nikromen** [comment]
right, I forgot about that, sorry

**Comment by nikromen** [comment]
done

**Review by praiskup — APPROVED** [review]
Seems OK, we talk about gitconfig file **in a mock chroot**. Thank you!

**Pull request merged**